server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java

```diff
@@ -308,6 +307,11 @@ private Map<String,String> conf(TCredentials credentials, AccumuloConfiguration
   @Override
   public Map<String,String> getConfiguration(TInfo tinfo, TCredentials credentials,
       ConfigurationType type) throws TException {
+    if (!(security.isSystemUser(credentials) || security.hasSystemPermission(credentials,
+        credentials.getPrincipal(), SystemPermission.SYSTEM))) {
```

Comment:
I'm not sure, but curious if this OR is necessary. I'd expect the system user to have system permission already. Also, because this is likely just checking the username, this could be bypassing authentication for the system user. If this is the case, this will need to be fixed, as well as the other occurrences of doing this elsewhere in this PR.

Reply:
Well, I can say that without this, things like DumpConfigIT, which use the Admin utility, failed.

Reply:
[isSystemUser](https://github.com/apache/accumulo/blob/main/server/base/src/main/java/org/apache/accumulo/server/security/SecurityOperation.java#L148-L152) just checks that the name of the token class in the API call is the same name as the token class used by the server. There is no "user" in this case.

Reply:
Right, so that's equivalent to comparing the user principal. The class name is basically the equivalent to the system user name.

But, the system user also has a password, and the authenticate method call was removed from line 296. We need to be very careful we're not bypassing the checking of the password for any RPC request that just masquerades as the system user, but doesn't actually authenticate as the system user.

The code highlighted here seems to only check the system user principal, and proceeds regardless of whether it authenticates. What I can't tell without taking a deeper look, is if the user has already been previously authenticated before reaching this line. If it has in every case, then this is fine... but if it hasn't, we still need to do that.

Reply:
> I'm not sure, but curious if this OR is necessary. I'd expect the system user to have system permission already. 

To answer your original question, if I remove the `security.isSystemUser(credentials)` predicate, then I get the following error in the Admin log for DumpConfigIT.

```
2022-10-06T11:23:06,542 [util.Admin] ERROR: Error USER_DOESNT_EXIST for user !SYSTEM - The user does not exist
org.apache.accumulo.core.client.AccumuloSecurityException: Error USER_DOESNT_EXIST for user !SYSTEM - The user does not exist
        at org.apache.accumulo.core.rpc.clients.TServerClient.execute(TServerClient.java:110) ~[classes/:?]
        at org.apache.accumulo.core.rpc.clients.ClientServiceThriftClient.execute(ClientServiceThriftClient.java:52) ~[classes/:?]
        at org.apache.accumulo.core.clientImpl.InstanceOperationsImpl.getSiteConfiguration(InstanceOperationsImpl.java:164) ~[classes/:?]
        at org.apache.accumulo.server.util.Admin.printConfig(Admin.java:604) ~[classes/:?]
        at org.apache.accumulo.server.util.Admin.execute(Admin.java:378) ~[classes/:?]
        at org.apache.accumulo.server.util.Admin.main(Admin.java:259) ~[classes/:?]
```

A warning is printed in the tablet server log, but nothing else
```
2022-10-06T11:23:06,519 [accumulo.audit] WARN : operation: failed; user: !SYSTEM; checking permission SYSTEM on !SYSTEM denied; exception: ThriftSecurityException(user:!SYSTEM, code:USER_DOESNT_EXIST)
```

I'm not quite sure of the exact code path that is raising this exception on the server side. If we are able to resolve this and remove the `security.isSystemUser(credentials)` predicate, then the authenticate call is made as part of the SecurityOperation.hasSystemPermission() call.